azure-pipelines-dev.yml

```diff
@@ -225,16 +245,24 @@ stages:
               summaryFileLocation: '$(System.DefaultWorkingDirectory)/**/coverage.xml'
               reportDirectory: '$(System.DefaultWorkingDirectory)/**/htmlcov'
 
+      # Runs pytest with Spark and Postgres enabled
       - job: comprehensive
         condition: eq(stageDependencies.scope_check.changes.outputs['CheckChanges.GEChanged'], true)
-        timeoutInMinutes: 120
+        strategy:
+          # This matrix is intended to split up our sizeable test suite into two distinct components.
+          # By splitting up slow tests from the remainder of the suite, we can parallelize test runs
+          # at the cost of an additional Azure worker.
+          #
+          # To work as intended, "standard" and "slow" should be equally performant.
+          matrix:
+            standard:
+              pytest_args: 'tests --ignore "tests/rule_based_profiler"'
+            slow:
+              pytest_args: 'tests/rule_based_profiler'
 
         services:
           postgres: postgres
 
-        variables:
-          GE_pytest_opts: '--postgresql --spark'
```

Comment:
Since these opts are consistent for all elements in the matrix, I've decided to hardcode.